### PR TITLE
Fix WASI API.

### PR DESCRIPTION
**Pull request by q82419**

### CODE REVIEW

**Comment by q82419** [comment]
@yanganto Please help to check the rust code. Thanks.

**Review by yanganto — APPROVED** [review]
LGTM.  Thanks.

File: lib/api/wasmedge.cpp [hunk 1]
```diff
@@ -2174,15 +2172,9 @@ WASMEDGE_CAPI_EXPORT void WasmEdge_ImportObjectInitWASI(
...
-  if (Dirs) {
-    for (uint32_t I = 0; I < DirLen; I++) {
-      DirVec.emplace_back(Dirs[I]);
-    }
-  }
   if (Preopens) {
     for (uint32_t I = 0; I < PreopenLen; I++) {
-      DirVec.emplace_back(std::string(Preopens[I]) + ":" +
-                          std::string(Preopens[I]));
+      DirVec.emplace_back(Preopens[I]);
```
Comment (hydai) [inline lib/api/wasmedge.cpp:2177]: We should have two different cases:

1. `PATH1:PATH2`: map `PATH1` to `PATH2`.
2. `PATH`: map `PATH` to `PATH`

However, this section cannot present these two behaviour.

**Review by hydai — CHANGES_REQUESTED** [review]

**Comment by hydai** [comment]
You should also update the changelog since this is a API breaking change.

File: lib/api/wasmedge.cpp (hunk 1 above)
Comment (q82419) [reply lib/api/wasmedge.cpp:2177]: This is actually handled in WASI:

https://github.com/WasmEdge/WasmEdge/blob/master/lib/host/wasi/environ.cpp#L58

Comment (hydai) [reply lib/api/wasmedge.cpp:2177]: I don't think so. Looks like the `environ.cpp` only handles the `PATH1:PATH2` part. It doesn't create the mapping of second one.

**Review by hydai — APPROVED** [review]

**Review by yanganto — APPROVED** [review]

**Pull request merged**